### PR TITLE
Implementation of the Barrel services for the MFT

### PR DESCRIPTION
**Pull request by rhelacon**
Implementation of the Barrel services following the suggestion from older pull request and working in the compatibility with the cmake changes

### CODE REVIEW

**Comment by rhelacon** [comment]
Can I get this reviewed? the error is not related, thanks

**Comment by rhelacon** [comment]
@shahor02 , @aphecetche

**Comment by shahor02** [comment]
please also squash the commits (alternatively can be squashed to 1st commit during merging)

File: Detectors/ITSMFT/MFT/base/include/MFTBase/Barrel.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  Barrel();
+  ~Barrel() = default;
+
+  TGeoVolumeAssembly* createBarrel();
+
+  // protected:
+  //  TGeoVolumeAssembly* mBarrel;
+
+ private:
+  ClassDef(Barrel, 1)
```
[the remark this answers is not in the document]
Comment (rhelacon) [reply Detectors/ITSMFT/MFT/base/include/MFTBase/Barrel.h]: ok, Thanks. I'll change that.

File: Detectors/ITSMFT/MFT/base/src/Barrel.cxx [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+#include "TGeoTube.h"
+#include "TGeoVolume.h"
+#include "TGeoXtru.h"
+#include "TMath.h"
+
+#include "MFTBase/Barrel.h"
+
+using namespace o2::mft;
+
+ClassImp(o2::mft::Barrel);
```
[the remark this answers is not in the document]
Comment (rhelacon) [reply Detectors/ITSMFT/MFT/base/src/Barrel.cxx]: ok

**Comment by rhelacon** [comment]
@shahor02 How can I squash the commits?

File: Detectors/ITSMFT/MFT/base/src/Barrel.cxx [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+    gGeoManager->MakeTube("Cone_Pipes", kMeAl, 0.3, 0.4, 82.25);
+  TGeoVolume* PipeBox0 =
+    gGeoManager->MakeBox("PipeBoxOut", kMeAl, 2.95, 0.45, 82.375);
+  TGeoVolume* PipeBox1 =
+    gGeoManager->MakeBox("PipeBoxInn", kMeAl, 2.85, 0.35, 82.375);
+  TGeoCompositeShape* ParallelPipeBox =
+    new TGeoCompositeShape("ParallelPipeBox", "PipeBoxOut - PipeBoxInn");
+  TGeoVolume* PipeBox = new TGeoVolume("PipeBox", ParallelPipeBox, kMeAl);
+
+  /*mm
```
[the remark this answers is not in the document]
Comment (rhelacon) [reply Detectors/ITSMFT/MFT/base/src/Barrel.cxx]: removed obselete comments

**Comment by shahor02** [comment]
@rhelacon do
git rebase -i HEAD~10 
then change for the commits you want to squash the keyword 'pick' to 's' or 'f', you will see the instructions.

**Comment by rhelacon** [comment]
@shahor02 
I followed your suggestions and squash the commits into 1, is now ok to be merged?, 
thanks!

**Pull request merged**